pkg/server/server.go

```diff
@@ -57,8 +57,14 @@ func (c *Server) watchOperatorResources(errChan chan error, ctx context.Context)
 		logger.WithError(err).Error("error setting queue policy")
 	}
 
-	logger.WithFields(logrus.Fields{"resources": c.Config.Resources}).Info("Watching")
-	go operators.Watch(ctx, corev1.NamespaceAll)
+	k8sNamespaceToWatch := c.Config.K8sNamespace
+	if k8sNamespaceToWatch == "" {
+		logger.WithFields(logrus.Fields{"resources": c.Config.Resources, "in namespace": "all"}).Info("Watching")
+		go operators.Watch(ctx, corev1.NamespaceAll)
+	} else {
+		logger.WithFields(logrus.Fields{"resources": c.Config.Resources, "in namespace": k8sNamespaceToWatch}).Info("Watching")
+		go operators.Watch(ctx, k8sNamespaceToWatch)
+	}
```

Comment:
I think this could be implemented a little simpler like so:

```go
logger.WithFields(logrus.Fields{"resources": c.Config.Resources, "namespace": k8sNamespaceToWatch}).Info("Watching")
go operators.Watch(ctx, k8sNamespaceToWatch)
```

`corev1.NamespaceAll` if I'm not wrong returns and empty string which is also what we're setting as the default. https://github.com/kubernetes/api/blob/master/core/v1/types.go#L30

Reply:
You're right. I debated that myself but wondered if it was a little clearer about what it's doing the way it is. But happy to make the change.